--- a/src/netcfg/proxy-netcfg-device.cpp
+++ b/src/netcfg/proxy-netcfg-device.cpp
@@ -206,7 +206,7 @@
             // Parse --dns server X address RESOLVER_ADDR...
             for (const auto &dns_addr : dnsopts.addresses)
             {
-                std::string addr = dns_addr.address.to_string();
+                std::string addr = dns_addr.address;
                 if ("UNSPEC" == addr)
                 {
                     log->LogError("DNS resolver setup: "
